views_api.py

```diff
@@ -80,76 +73,107 @@ async def api_delete_relay(relay: Relay):  # type: ignore
-        signature=event.sig,
-    )
-    client.relay_manager.publish_event(nostr_event)
-
-
-@nostrclient_ext.post("/api/v1/filters")
-async def api_subscribe(filters: Filters):
-    nostr_filters = init_filters(filters.__root__)
-
-    return EventSourceResponse(
-        event_getter(nostr_filters),
-        ping=20,
-        media_type="text/event-stream",
-    )
-
-
-@nostrclient_ext.websocket("/api/v1/filters")
-async def ws_filter_subscribe(websocket: WebSocket):
-    await websocket.accept()
-    while True:
-        json_data = await websocket.receive_text()
-        try:
-            data = json.loads(json_data)
-            filters = data if isinstance(data, list) else [data]
-            filters = [Filter.parse_obj(f) for f in filters]
-            nostr_filters = init_filters(filters)
-            async for message in event_getter(nostr_filters):
-                await websocket.send_text(message)
-
-        except Exception as e:
-            logger.warning(e)
-
-
-def init_filters(filters: List[Filter]):
-    filter_list = []
+def marshall_nostr_filters(data: Union[dict, list]):
+    filters = data if isinstance(data, list) else [data]
+    filters = [Filter.parse_obj(f) for f in filters]
+    filter_list: list[NostrFilter] = []
     for filter in filters:
         filter_list.append(
             NostrFilter(
-                event_ids=filter.ids,
+                event_ids=filter.ids,  # type: ignore
                 kinds=filter.kinds,  # type: ignore
-                authors=filter.authors,
-                since=filter.since,
-                until=filter.until,
-                event_refs=filter.e,
-                pubkey_refs=filter.p,
-                limit=filter.limit,
+                authors=filter.authors,  # type: ignore
+                since=filter.since,  # type: ignore
+                until=filter.until,  # type: ignore
+                event_refs=filter.e,  # type: ignore
+                pubkey_refs=filter.p,  # type: ignore
+                limit=filter.limit,  # type: ignore
             )
         )
-
-    nostr_filters = NostrFilters(filter_list)
-    subscription_id = urlsafe_short_hash()
-    client.relay_manager.add_subscription(subscription_id, nostr_filters)
-
-    request = [ClientMessageType.REQUEST, subscription_id]
-    request.extend(nostr_filters.to_json_array())
-    message = json.dumps(request)
-    client.relay_manager.publish_message(message)
-    return nostr_filters
-
-
-async def event_getter(nostr_filters):
+    return NostrFilters(filter_list)
+
+
+async def add_nostr_subscription(json_str):
+    """Parses a (string) request from a client. If it is a subscription (REQ), it will
+    register the subscription in the nostr client library that we're using so we can
+    receive the callbacks on it later"""
+    json_data = json.loads(json_str)
+    assert len(json_data)
+    if json_data[0] == "REQ":
+        subscription_id = json_data[1]
+        fltr = json_data[2]
```

Comment:
- why bother processing the filters, just forward them?

Reply:
That's how the library works 🤷: https://github.com/jeffthibault/python-nostr

Not important enough to justify a change of the lib imo.